### PR TITLE
Print out parsing errors if we're going to crash

### PR DESCRIPTION
**Pull request by jyn514**
Note that this doesn't do anything about lexing errors because the parser doesn't know the type of the preprocessor.

### CODE REVIEW

File: src/parse/mod.rs [hunk 1]
```diff
@@ -184,6 +184,16 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
                 "fatal: maximum recursion depth exceeded ({} > {})",
                 depth, MAX_DEPTH
             );
+            if !self.error_handler.is_empty() {
+                println!("pending errors:");
+                // we're about to crash, no point worrying about performance
+                for error in self.error_handler.clone() {
```
Comment (pythondude325) [inline src/parse/mod.rs]: Why not simple avoid this clone?

Comment (jyn514) [reply src/parse/mod.rs]: `recursion_check` takes `&self`, not `&mut self`.

Comment (pythondude325) [reply src/parse/mod.rs]: But why not clone to a temp variable and use that error handler twice?

File: src/parse/mod.rs [hunk 2]
```diff
@@ -184,6 +184,16 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
...
+                    println!("- error: {}", error.data);
+                }
+                for warning in self.error_handler.clone().warnings {
+                    println!("- warning: {}", warning.data);
+                }
```
Comment (pythondude325) [inline src/parse/mod.rs]:
```suggestion
                // we're about to crash, no point worrying about performance
                let error_handler = self.error_handler.clone();
                for error in error_handler {
                    println!("- error: {}", error.data);
                }
                for warning in error_handler.warnings {
                    println!("- warning: {}", warning.data);
                }
```

**Pull request merged**